fc/jira/task.py

```diff
@@ -4,77 +4,9 @@
 from ..auth.auth import Auth
 from typing import Optional
 from ..exceptions.task_exception import TaskException
+from .issue import Issue
 
-
-class Task:
-    api_url = 'https://jira.cms.gov/rest/api/2/issue/'
-    base_url = 'https://jira.cms.gov/browse/{}'
-    transition_id_for_triage_ready = '11'
-    transition_id_for_start_progress = '21'
-    transition_id_for_ready_for_refinement = '201'
-    transition_id_for_refined = '211'
-    transition_id_for_task_ready = '221'
-    issue_assigned_sprint_field = 'customfield_10005'
-
-    def __init__(self):
-        self.title = None
-        self.description = None
-        self.id = None
-        self.url = None
-        self.type = None
-        self.state = None
-        self.auth = None
-
-    def from_json(self, json: dict, auth: Auth):
-        self.title = json['fields']['summary']
-        self.description = json['fields']['description']
-        self.id = json['key']
-        self.url = self.base_url.format(json['key'])
-        self.type = json['fields']['issuetype']['name']
-        self.state = json['fields']['status']['name']
-        self.auth = auth
-
-        return self
-
-    def from_args(self, title: str, description: str, auth: Auth):
-        self.title = title
-        self.description = description
-        self.auth = auth
-
-        return self
-
-    def create(self):
-        json = {
-            'fields': {
-                'project': {
-                    'key': 'QPPFC'
-                },
-                'summary': self.title,
-                'description': self.description,
-                'components': [{
-                    'name': 'Foundational'
-                }]
-            }
-        }
-
-        self._extra_json_for_create(json)
-
-        response = requests.post(self.api_url, json=json,
-                                 auth=HTTPBasicAuth(self.auth.username(), self.auth.password()))
-        response.raise_for_status()
-
-        response_json = response.json()
-
-        self.id = response_json['key']
-        self.url = self.base_url.format(self.id)
-
-        return self.id, self.url
-
-    def _extra_json_for_create(self, existing_json: dict):
-        raise NotImplementedError
-
-    def type_str(self) -> str:
-        raise NotImplementedError
+class Task(Issue):
 
     def transition(self, state: str):
```

Comment:
`transition` seems to be `Task`-specific or a function that will get implemented with class-specific logic, so leave it?

Reply:
Yeah, let's keep it in `Task` since we won't transition any generic Jira issue.